deepspeed/runtime/engine.py

```diff
@@ -467,6 +469,13 @@ def __getattr__(self, name):
             return getattr(self, name)
         elif name in dir(_module):
             return getattr(_module, name)
+        elif isinstance(_module, CompiledModuleWrapper):
```

Comment:
This condition is necessary to make custom functions of the wrapped module work.

Assume we have the compiled module `cm`, its wrapper `w` (`CompiledModuleWrapper`), and the deepspeed engine `engine`. We want to call a custom function `f` of `cm` by `engine.f()`. However, `dir(_module)` doesn't return `f`. So `engine.f()` would fail without this condition though `w.f()` works.

I'm not sure this is the best design for the purpose. I would appreciate it if you could give my any suggestion.

Reply:
Will `hasattr(_module, name)` work for both cases of compiled and not-compiled module? In other words, for non-compiled module, can we use `hasattr(_module, name)` instead of `name in dir(_module)`?